src/dfx/src/commands/extension/install.rs

```diff
@@ -0,0 +1,52 @@
+use crate::lib::environment::Environment;
+use crate::lib::error::DfxResult;
+use crate::lib::extension::manifest::fetch_dfinity_extension_manifest;
+
+use std::io::Cursor;
+
+use clap::Parser;
+use semver::{Prerelease, VersionReq};
+
+
+#[derive(Parser)]
+pub struct InstallOpts {
+    /// Specifies the name of the extension to install.
+    extension_name: String,
+}
+
+pub fn exec(env: &dyn Environment, opts: InstallOpts) -> DfxResult<()> {
+    let manifest = fetch_dfinity_extension_manifest();
+
+    let mut v = env.get_version().clone();
+    // Remove the prerelease tag, if any. This is because prerelease tags
+    // should not be allowed in extension manifests, and semver crate
+    // won't match a semver with a prerelease tag against a semver without.
+    v.pre = Prerelease::EMPTY;
+
+    for (dfx_version, manifests) in manifest.unwrap().0.iter() {
+        let sv_test = VersionReq::parse(dfx_version);
+        if sv_test.unwrap().matches(&v) {
```

Comment:
How about some types for these, something like `ExtensionManifest`, `ExtensionDefinition`, `ExtensionCache`, `Extension`:

```
let manifest = ExtensionManifest::fetch()?; // or ::new()
let extension_definition = manifest.find(extension_name, dfx_version)?;
let extension_cache = ExtensionCache::new(dfx_version);
let extension = extension_cache.install(extension_definition)?;
```

Reply:
awesome

Reply:
I did my own spin of the suggested `ExtensionCache` - instead of having it as a struct, I made it as a `trait ExtensionsManager`, let me know how you like it